tests/eosio.system_tests.cpp

```diff
@@ -1998,6 +1998,30 @@ BOOST_FIXTURE_TEST_CASE(voter_gbm_pay, eosio_system_tester, * boost::unit_test::
    }
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(burning_preminted, eosio_system_tester, * boost::unit_test::tolerance(1e-10)) try {
+   cross_15_percent_threshold();
+
+   create_account_with_resources(N(user11111111), config::system_account_name, core_sym::from_string("100.0000"), false,
+                                 core_sym::from_string("10.0000"), core_sym::from_string("10.0000"));
+
+   // This transer creates a sub_balance for genesis.wax account
+   transfer( N(eosio), N(genesis.wax), core_sym::from_string("100000000.0000"), N(eosio) );
+
+   uint64_t nonce = 1;
+   asset genesis_tokens_user1{core_sym::from_string("50000000.0000")};
+   awardgenesis(N(user11111111), genesis_tokens_user1, nonce);
+
+   produce_block( fc::days(1) ); //Wait until July 1st
+   produce_block( fc::days(1096 / 2) );
```

Comment:
replacing this or another test where this is like `produce_block( fc::days(1096 / 3) ) is probably a good idea to make sure we don't only succeed due to some symmetry in the math

Reply:
Also, we should probably test the case where we go past the 3 year gbm period with something like `produce_block( fc::days(1096 * 2) )`

Reply:
If that minimizer guard is in place, it should result in 0 being sent back rather than a negative no? In fact we may just need to check that the value being sent back is > 0 rather than messing with the minimizer

Reply:
Yes, once we have passing these unit tests (including the one failing), I'll add a bunch of more tests.